src/xtgeo/cube/cube1.py

```diff
@@ -1054,15 +1081,15 @@ def _ensure_correct_values(self, values):
 
         if isinstance(values, numbers.Number):
             self._values = np.zeros(self.dimensions, dtype=np.float32) + values
-            return
 
-        if isinstance(values, np.ndarray):
+        elif isinstance(values, np.ndarray):
             values = values.reshape(self.dimensions)
 
             if not values.data.c_contiguous:
                 values = np.ascontiguousarray(values)
+            self._values = values
 
-        if isinstance(values, (list, tuple)):
-            values = np.array(values, dtype=np.float32).reshape(self.dimensions)
+        elif isinstance(values, (list, tuple)):
+            self._values = np.array(values, dtype=np.float32).reshape(self.dimensions)
 
-        self._values = values
+        self._values = self._values.astype(np.float32)
```

Comment:
Maybe you could `.astype(np.float32)` in the last statement of first `elif`, and put this trailing setting in an `else`?